### PR TITLE
Extending rbac

### PR DESCRIPTION
**Pull request by jasonenglish2039**
In the "Zowe Application Framework configuration" section, added a sub-section "Applying role-based access control to application services".

### CODE REVIEW

**Comment by jasonenglish2039** [comment]
Made a mistake. The pull request should be to doc-staging, not master.

File: docs/user-guide/mvd-configuration.md [hunk 1]
```diff
@@ -292,7 +292,90 @@ cp "//'[output_dataset_name]'" 'zlux-app-server/deploy/instance/ZLUX/serverConfi
 }
 ```
 
-### Enabling tracing
+## Applying role-based access control to application services
+
+To use role-based access control (RBAC) for Zowe service endpoints, enable RBAC for Zowe, and then use a z/OS security product such as RACF to map roles and authorities to Zowe service endpoints.
+
+After you configure RBAC, when a Zowe application wants to access an endpoint the application server sends the current user ID and endpoint identifier URL to the ZSS server. The ZSS server queries the z/OS System Authorization Facility (SAF), which determines whether the access is authorized.
```
Comment (fkovinAtRocket) [inline docs/user-guide/mvd-configuration.md]: "sends [...] and endpoint identifier URL to the ZSS server" -- this is not exactly correct. Not sure if the doc needs to be that detailed. I would say something like "the application server communicates with the ZSS server to find out if the current user ID  can access the endpoint identifier URL"

File: docs/user-guide/mvd-configuration.md [hunk 2]
```diff
@@ -292,7 +292,90 @@ cp "//'[output_dataset_name]'" 'zlux-app-server/deploy/instance/ZLUX/serverConfi
...
+
+For example, to issue a POST request to the service endpoint documented above using the default ==Zowe Application Server?== instance, users must have READ access to the following SAF profile:
+
+`ZLUX.DEFAULT.SVC.ORG_ZOWE_FOO.BAZ.POST.USERS.FRED`
+
+Note: "DEFAULT" is the default Zowe instance ID.
+
+Configuration endpoints have SAF profile names with this format:
+
+`<product>.<instance id>.**CFG.<pluginid_with_underscores>.<service>.<HTTP method>.<url with forward slashes '/' replaced by periods '.'>`
```
Comment (fkovinAtRocket) [inline docs/user-guide/mvd-configuration.md]: 1. not sure why it's saying **CFG: it's just CFG
2. I'd mention the scope here

File: docs/user-guide/mvd-configuration.md [hunk 3]
```diff
@@ -292,7 +292,90 @@ cp "//'[output_dataset_name]'" 'zlux-app-server/deploy/instance/ZLUX/serverConfi
...
+1. Open the Zowe Application Server configuration JSON file. In the default server instance, the configuration file is `/zlux-app-server/config/zluxserver.json`.
+2. In the `dataserviceAuthentication` object, add `"rbac": true`.
+
+### Configuring basic authorization
+
+We recommend configuring some basic authorization:
+
+- To give administrators access to everything in Zowe, create a `ZLUX.**` profile and give them UPDATE access to it.
+- To give non-administrators basic access to the site and product, create a `ZLUX.*.ORG_ZOWE_*`  profile and give them READ access to it.
+- To prevent non-administrators from configuring endpoints at the product and instance levels, create a ?? profile and give them no access to it.
```
Comment (fkovinAtRocket) [inline docs/user-guide/mvd-configuration.md]: `ZLUX.DEFAULT.CFG.ORG_ZOWE_FOO.**`

File: docs/user-guide/mvd-configuration.md [hunk 4]
```diff
@@ -292,7 +292,90 @@ cp "//'[output_dataset_name]'" 'zlux-app-server/deploy/instance/ZLUX/serverConfi
...
+- To give non-administrators all access to user, create a ?? profile and give them ?? access to it.
```
Comment (fkovinAtRocket) [inline docs/user-guide/mvd-configuration.md]: `ZLUX.DEFAULT.CFG.ORG_ZOWE_FOO.*.USER.**`

File: docs/user-guide/mvd-configuration.md [hunk 5]
```diff
@@ -292,7 +292,90 @@ cp "//'[output_dataset_name]'" 'zlux-app-server/deploy/instance/ZLUX/serverConfi
...
+
+### Creating generic authorization
```
Comment (fkovinAtRocket) [inline docs/user-guide/mvd-configuration.md]: I would mention this earlier, given that we've already recommended creating generic profiles (i.e. ones that contain * or **)

File: docs/user-guide/mvd-configuration.md [hunk 6]
```diff
@@ -292,7 +292,90 @@ cp "//'[output_dataset_name]'" 'zlux-app-server/deploy/instance/ZLUX/serverConfi
...
+
+Some endpoints can generate unlimited numbers of URLs. To apply RBAC to them you must create generic profiles. For example, an endpoint that performs a DELETE action on any file would generate a different URL for each file and users can create an unlimited number of files. So you must create a generic profile to handle access, such as:
+
+`ZLUX.DEFAULT.SVC.ORG_ZOWE_FOO.BAZ.DELETE.**` 
+
+You can create generic profile names using wildcards, such as asterisks (*). For information on generic profile naming, see [IBM documentation](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.1.0/com.ibm.zos.v2r1.icha100/egnoff.htm).
+
+### Limiting endpoint URL length
```
Comment (fkovinAtRocket) [inline docs/user-guide/mvd-configuration.md]: And this part is for the app authors, not security admins

**Pull request closed without merging**